test/e2e/specs/editor/various/splitting-merging.spec.js

```diff
@@ -0,0 +1,404 @@
+<p>Second</p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should remove empty paragraph block on backspace', async ( {
+		editor,
+		page,
+	} ) => {
+		// Regression Test: In a sole empty paragraph, pressing backspace
+		// should remove the block.
+		//
+		// See: https://github.com/WordPress/gutenberg/pull/8306
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.press( 'Backspace' );
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe( `` );
+	} );
+
+	test( 'should remove at most one paragraph in forward direction', async ( {
+		editor,
+		page,
+	} ) => {
+		// Regression Test: A forward delete on empty RichText previously would
+		// destroy two paragraphs on the dual-action of merge & remove.
+		//
+		// See: https://github.com/WordPress/gutenberg/pull/8735
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.type( 'First' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Delete' );
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:paragraph -->
+<p>First</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>Second</p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should ensure always a default block', async ( {
+		editor,
+		page,
+	} ) => {
+		// Feature: To avoid focus loss, removal of all blocks will result in a
+		// default block insertion at the root. Pressing backspace in a new
+		// paragraph should not effectively allow removal. This is counteracted
+		// with pre-save content processing to save post consisting of only the
+		// unmodified default block as an empty string.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/9626
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.press( 'Backspace' );
+
+		// There is a default block and post title:
+		expect(
+			await page.locator( '.block-editor-block-list__block' )
+		).toBeDefined();
+
+		// But the effective saved content is still empty:
+		expect( await editor.getEditedPostContent() ).toBe( '' );
+
+		// And focus is retained:
+		await expect(
+			page.locator( 'role=document[name=/Empty block/i]' )
+		).toBeFocused();
+	} );
+
+	test( 'should undo split in one go', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '12' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'z' );
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
```

Comment:
Suggestion (non-blocking): When padding post content to inline snapshots, you can also use `expect.poll` to auto-wait for assertion. It can be used in most of the tests in this suite.

```
await expect.poll( editor.getEditedPostContent ).toBe( '' )
```

Note: The method does work with `toMatchSnapshot` assertions.

Reply:
@pooja-muchandikar, sorry for miss understanding. There's no need to convert inline snapshot assertions to `toMatchSnapshot`.

For small text nodes, I would always recommend inline snapshots.

Reply:
Hi @Mamaduka,

I tried the above suggestion but somehow its failing in local, so that's the reason I have added `toMatchSnapshot` assertion. 

Should I revert it?

Reply:
That's odd. Thanks for testing it locally. There's no need to revert. I will merge the PR once all checks are green.

Thanks again for the contribution!

Reply:
Thank you!